src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -13088,7 +13088,7 @@ bool writeDataSizeToSnapshot
                 ||
                 (buff.Length >= offst + len)
                 ||
-                (buff.Length == (startOffsetByteCount >> 1) + 1),
+                (buff.Length >= (startOffsetByteCount >> 1) + 1),
```

Comment:
This change simply allows the target buffer we're copying into to be a bit larger than we actually need. This is required so that we can make the later change which grows the buffer more than we need.